### PR TITLE
[Feature] Add support for multi-task CD models

### PR DESCRIPTION
**Pull request by Bobholamovic**
1. Add one SOTA muti-task CD model, ChangeStar (the paper was published in ICCV 2021).
2. The new `multitask_loss_computation` function calculates the losses between multiple output logits and multiple masks.
3. The new field `aux_masks` in the sample dict is given such a name that it can be used for different tasks. For example, an edge mask can be stored as a `aux_mask` and used in a semantic segmentation task. Transformation operations on `aux_masks` have been supported.
4. A new protocol has been proposed for multi-task CD models. To enable the feature of multi-task training, an `nn.Layer` object should have an attribute named `USE_MULTITASK_DECODER` and set the value of it to `True`. Also, the object should have an attribute named `OUT_TYPES`, which is a tuple containing `LabelType` enums, each element in this tuple denoting the label type of the corresponding element in the output list of the `forward` method during model training.
5. Now `CDDataset` accepts an optional argument `with_seg_labels` and supports either 3 or 5 items in a line of the `file_list`.

### CODE REVIEW

**Review by geoyee — APPROVED** [review]

**Review by LutaoChu — CHANGES_REQUESTED** [review]
需要解决一下冲突

**Comment by Bobholamovic** [comment]
> 需要解决一下冲突

冲突已解决。因为merge不小心误操作为rebase所以在自己的分支forcely push了一次，不过没有影响到主分支。此外，新增了两个小功能：

- 变化检测数据集现在能够自动对标签进行二值化处理。在数据集中添加该功能而不是将其实现为一个transform，是考虑到在eval阶段，变化检测任务的transform似乎执行了apply_im_only模式，因此对标签的处理被绕过。
- 添加了随机交换两个时相影像的transform，这在变化检测模型的训练中常用。
